### ISSUE TITLE
automatically do npm install in build

### ISSUE THREAD

**Issue by WayneUong** [zodern/meteor-up#474]
Can we have the option of automatically running `meteor npm install` during bundling?

**Comment by zodern** [comment]
Could you please explain why you need this? What are you trying to do or what that is not working?

**Comment by WayneUong** [comment]
We have a git repo setup for continuous deployment with `mup` on a remote server. Any time the repo is updated, a deployment process is automatically triggered. But every time the `packages.json` list is updated, we have to manually ssh to the server and run `meteor npm install` before `mup` can bundle the repo successfully with all the latest npm packages.

**Comment by zodern** [comment]
Are you able to automatically always run `meteor npm install` before `mup deploy`?

**Comment by WayneUong** [comment]
We can, but then we will have to inject a middle-man script just for that purpose (the other parts are all outsourced). So I'm wondering if it can be done within `mup`.

**Comment by jessb321** [comment]
Yeah, I noticed this, since I do all the deployment myself, I just added meteor npm install to my script after git pulling. Not a big issue, but would be nice to have.

**Comment by shadowcodex** [comment]
Would running `meteor npm install` in some way cause an unexpected effect against a meteor project? Like if someone removed manually a package but left it in packages.js?

Would it be best to outline how to add meteor up to a CI system?

**Comment by zodern** [comment]
> Would it be best to outline how to add meteor up to a CI system?

We do need to do that.

@WayneUong would a predeploy hook as proposed in #65 work?

**Comment by zodern** [comment]
Hooks will be available in Mup 1.3. I would appreciate help testing it.

You can install the beta with `npm i -g mup@next`. Then, in your config add:
```js
{
  //... rest of config

  hooks: {
    'pre.deploy': { localCommand: 'npm install' }
  }
}
```

**Comment by znat** [comment]
Hi @WayneUong , I was wondering if you could briefly describe how you setup CD?
